docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,605 @@ under the License.
+is a [task slot](#task-slot). 
+
+#### Timer
+
+Timers allow applications to react to changes in [processing time](#processing-time) and in [event time](#event-time).
+There are at most one timer per key and per second.
+
+Timers are fault-tolerant and checkpointed along with the state of the application. In case of a failure 
+recovery or when starting an application from a [savepoint](#savepoint), timers are restored.
 
 #### Transformation
 
-A Transformation is applied on one or more data streams or data sets and results in one or more
-output data streams or data sets. A transformation might change a data stream or data set on a
-per-record basis, but might also only change its partitioning or perform an aggregation. While
-[Operators](#operator) and [Functions](#function) are the "physical" parts of Flink's API,
-Transformations are only an API concept. Specifically, most transformations are
-implemented by certain [Operators](#operator).
+A transformation is applied to one or more [DataStreams](#datastreams) and results in one or more 
+output DataStreams. A transformation might change a DataStream on a [per-record](#record) basis, but 
+might also only change its [partitioning](#partition) or perform an [aggregation](#aggregation). While 
+[operators](#operator) and [functions](#function) are the "physical" parts of Flink's API, transformations 
+are an API concept. Specifically, most transformations are implemented by certain [operators](#operator).
+
+#### Tumbling Window
+
+This is a [window](#window) that groups elements by a specified window size. Tumbling windows have a 
+fixed size and do not overlap. For example, if you specify a tumbling window with a size of 5 minutes, 
+the current window will be evaluated and a new window will be started every five minutes.
+
+#### Tuple
+
+A composite data type that has a finite ordered list of immutable elements. 
+
+#### Unbounded streams
+
+Unbounded [DataStreams](#datastream) have a start but no defined end. They do not terminate, provide 
+data as it is generated, and must be continuously processed. 
+
+#### (User-Defined) Functions
+
+Functions are implemented by the user and encapsulate the application logic of a [Flink application](#(flink)-application). 
+Most functions are wrapped by a corresponding [operator](#operator).
+
+#### User-Defined Aggregate Function (UDAF)
+
+This type of user-defined function aggregates multiple values into a single value.
+
+#### User-Defined Scalar Function (UDSF)
+
+This type of user-defined function maps zero, one, or more [scalar](#scalar) values to a new scalar value.
+
+#### User-Defined Table-valued Function (UDTF)
+
+This type of user-defined function uses zero, one, or multiple [scalar](#scalar) values as input parameters 
+(including variable-length parameters). A UDTF returns any number of rows, rather than a single value. 
+The returned rows can consist of one or more columns.
```

Comment:
These are SQL specific, maybe in the _(User-Defined) Functions_ entry above just quote them. e.g.

> In Table API/SQL the terms User-Defined Function (UDF), User-Defined Aggregate Function (UDAF), User-Defined Scalar Function (UDSF) and User-Defined Table-valued Function (UDTF) are often used to refer to user-defined functions

And then link to https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/functions/udfs/ where we explain everything about it

Reply:
You can also have a Python UDF, right? So not all UDFs are SQL specific, or do you mean something else?

Reply:
UDF is specific to our relational apis, that is sql and table (and python table)